manifests/fragment.pp

```diff
@@ -68,6 +68,7 @@
 
   $safe_name        = regsubst($name, '[/:\n]', '_', 'GM')
   $safe_target_name = regsubst($target, '[/:\n]', '_', 'GM')
+  $safe_order       = regsubst($order, '[/:\n]', '_', 'GM')
```

Comment:
@tpdownes the code behaves differently depending on the sorting type [here](https://github.com/puppetlabs/puppetlabs-concat/blob/master/files/concatfragments.rb#L124-L141)

I think my main concern in this is around the use of the underscore, in the case where the ordering is set to numeric and there are invalid characters in the order, it will only sort on the order up to the first invalid character. Could you update the `regsubst` to replace it with a dash maybe instead?